### ISSUE TITLE
lerna version gets confused by non-lerna tags

### ISSUE THREAD

**Issue by fabb** [lerna/lerna#2377]
## Expected Behavior
* Given a repo that contains both lerna packages, and a non-lerna package with its own top-level subfolder and package.js
* The non-lerna package also creates git tags on new releases (the lerna package tags are prefixed with `@my-group/` while the non-lerna tags are prefixed with `legacy/`)
* `lerna version` should ignore git tags of the non-lerna package and being able to find and tag changed packages independently of git tags of the non-lerna package

## Current Behavior
When the latest tag on the `master` branch is from the non-lerna package, and after that no changes have been made to the lerna packages, lerna will output that there are no changes:
```
lerna info Looking for changed packages since legacy/0.35.3
lerna info No changed packages found
```

## Possible Solution
Ignore tags that have a non-matching name.

## Steps to Reproduce (for bugs)
1. Create a lerna repo with some packages and make a release
2. Make changes to the lerna packages and commit
3. Create a manual tag for the latest commit
4. Try to run `lerna version` and observe that it does not recognize the changes from 2.

<details><summary>lerna.json</summary><p>

```json
{
  "packages": [
    "packages/*"
  ],
  "version": "independent"
}
```
Note that the non-lerna package is NOT contained in the `packages/` folder.
</p></details>

## Context
We introduced lerna later in our repository and created new separate published packages for the lerna packages, and use a starvation strategy for the old legacy package.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.19.0 |
| `npm --version`  | 6.12.0 |
| `node --version` | v12.13.0 |

| OS | Version |
| --- | --- |
| macOS Catalina | 10.15.1 (19B88)|

**Comment by fabb** [comment]
I cannot even use `--force-publish` because that option is broken:
```
10:32:45,128 WARN  - #28 1.692 lerna WARN force-publish core
10:32:45,128 WARN  - #28 1.692 lerna WARN force-publish icons
10:32:45,128 WARN  - #28 1.692 lerna info Looking for changed packages since legacy/0.35.3
10:32:45,128 WARN  - #28 1.715 lerna success No changed packages to publish
```

**Comment by evocateur** [comment]
> I cannot even use `--force-publish` because that option is broken

The arguments to `--force-publish` are package names (or globs), I highly doubt you own the `core` or `icons` package.

**Comment by fabb** [comment]
I know these are globs, and yes, these are my packages. The globs work fine.

**Comment by evocateur** [comment]
Ok, just covering the bases. Thank you.

As for the underlying issue, I'm not sure Lerna can support this kind of situation. To some degree, choosing to use Lerna in a monorepo is also choosing to cede control over the generation of annotated release tags. The clearest solution here is either to incorporate the "legacy" package as a proper Lerna-managed package, or to remove that package from the monorepo entirely.

**Comment by fabb** [comment]
Yeah we thought about integrating it into lerna, held off yet because of migration effort. Couldn‘t lerna just consider tags that match the package org prefix, and ignore other tags?

**Comment by evocateur** [comment]
On the other hand, hacking around randomly I think I have a decent approach that simplifies a few other tag-related methods.

**Comment by fabb** [comment]
Sounds good 😅

**Comment by evocateur** [comment]
The changes have exposed some weaknesses in existing tests, so it's taking a little longer than anticipated. Yay tests?

**Comment by evocateur** [comment]
So I fixed the bad tests in`master`, but wasn't able to get to a complete solution on the tag filtering. I agree that it's valuable to have, but as I indicated in the commit that references this issue, I still haven't addressed the `collectUpdates()` usage here: https://github.com/lerna/lerna/blob/167d6e3abd81af1f8bed587f649b93016c76dc20/core/filter-options/lib/get-filtered-packages.js#L60

Without filtering that execution, almost any command outside of `lerna publish`/`lerna version` in a repo where external annotated git tags exist would likely be incorrect. Additional tests are needed to guard against regression, too.

**Comment by fabb** [comment]
Sounds like a lot of work, thank you for investing the time and providing this helpful tool!

**Comment by blakemann** [comment]
@evocateur I'm also currently a bit snagged by this at the moment, as our organization is transitioning from one release process to another right now, and it might be a while before we can stop creating any non-lerna tags in the repo.

I've tested out the code in your WIP commits, and it seems to work for the use cases I'd need (the version/publish commands are really the only ones we actively need to use). Is there any chance that could be making its way to the main lerna repo soon, despite not covering the `get-filtered-options.js` usage? A partial fix may still be more valuable than nothing!

**Comment by evocateur** [comment]
I still feel that the WIP branch provides important functionality, but I can't give any timeline on completion. Using lightweight tags for the "external" tags (instead of annotated tags) is the best workaround I can think of, right now.
